pkg/runtime/host.go

```diff
@@ -8,25 +8,37 @@ import (
 )
 
 const (
+	// HostIPEnvVar is the environment variable to override host's chosen IP address.
 	HostIPEnvVar = "DAPR_HOST_IP"
 )
 
+// GetHostAddress selects a valid outbound IP address for the host.
 func GetHostAddress() (string, error) {
 	if val, ok := os.LookupEnv(HostIPEnvVar); ok && val != "" {
 		return val, nil
 	}
 
-	addrs, err := net.InterfaceAddrs()
+	// Use udp so no handshake is made.
+	// Any IP can be used, since connection is not established, but we used a known DNS IP.
+	conn, err := net.Dial("udp", "8.8.8.8:80")
```

Comment:
8.8.8.8 is google one :) but it may not work inside intranet if it doesn't have the internet. Do we have better way ?

Reply:
I guess we are all ignoring code comments. See the comments above, please. This DOES NOT stablish a connectionl This is UDP. See my reply on this below.

Reply:
Resolving this as a duplicate of thread below.